File: element-common/src/main/java/io/github/solusmods/eternalcore/element/impl/network/c2s/ClientAccess.java
package io.github.solusmods.eternalcore.element.impl.network.c2s;

import io.github.solusmods.eternalcore.element.impl.ElementsStorage;
import lombok.AccessLevel;
import lombok.NoArgsConstructor;
import net.minecraft.server.level.ServerPlayer;

@NoArgsConstructor(access = AccessLevel.PRIVATE)
public class ClientAccess {

    static void handle(SyncElementsStoragePayload packet, ServerPlayer player) {
        player.eternalCore$getStorageOptional(ElementsStorage.getKey()).
                ifPresent(storage -> storage.load(packet.data()));
    }
}

File: element-common/src/main/kotlin/io/github/solusmods/eternalcore/element/impl/network/c2s/ClientAccess.kt
package io.github.solusmods.eternalcore.element.impl.network.c2s

import io.github.solusmods.eternalcore.element.impl.ElementsStorage
import io.github.solusmods.eternalcore.storage.api.StorageKey
import net.minecraft.server.level.ServerPlayer

object ClientAccess {
    fun handle(packet: SyncElementsStoragePayload, player: ServerPlayer?) {
        player!!.getStorageOptional(ElementsStorage.key as StorageKey<ElementsStorage?>?).ifPresent {
            storage: ElementsStorage? -> storage!!.load(packet.data!!)
        }
    }
}
